storage/raftstorage.go

```diff
@@ -206,6 +166,13 @@ func (r *Range) FirstIndex() (uint64, error) {
 	return ts.Index + 1, nil
 }
 
+// loadAppliedIndex retrieves the applied index from storage.
+func (r *Range) loadAppliedIndex() (uint64, error) {
+	return loadAppliedIndex(r.rm.Engine(), r.Desc().RaftID)
+}
+
+// loadAppliedIndex retrieves the applied index for the given
+// raftID from the supplied engine.
 func loadAppliedIndex(eng engine.Engine, raftID int64) (uint64, error) {
```

Comment:
I think we should just have `(*Range) loadAppliedIndex(eng engine.Engine) (uint64, error)`. We shouldn't have to pass in the RaftID since the range always knows it, but we should require the caller to pass the engine so you have to think about whether you want to use r.rm.Engine() or a batch/snapshot.